apps/storage/crud.py

```
@@ -468,6 +468,40 @@ async def get_pending_events(
+        session: AsyncSession,
+        *,
+        external_id: str,
+        tenant_id: uuid.UUID
+    ) -> Optional[Event]:
+        """Get event by external_id (for duplicate detection)."""
+        result = await session.execute(
+            select(Event).where(
+                and_(
+                    Event.external_id == external_id,
+                    Event.tenant_id == tenant_id,
+                    Event.is_deleted == False
```

Ruff (0.13.3)

Avoid equality comparisons to `False`; use `not Event.is_deleted:` for false checks

Replace with `not Event.is_deleted`

(E712)